component/azstorage/block_blob.go

```diff
@@ -85,11 +85,16 @@ func (bb *BlockBlob) Configure(cfg AzStorageConfig) error {
 	bb.Config = cfg
 
 	bb.blobAccCond = azblob.BlobAccessConditions{}
-	bb.blobCPKOpt = azblob.ClientProvidedKeyOptions{}
+	bb.blobCPKOpt = azblob.ClientProvidedKeyOptions{
+		EncryptionKey:       &bb.Config.cpkEncryptionKey,
+		EncryptionKeySha256: &bb.Config.cpkEncryptionKeySha256,
```

Comment:
CPK also has an option of sending scope in the parameter. If we make this change to go to main line then there will be other customers trying to use it and it will fail when they use scope so better to add that option here itself.

Reply:
Looked at the latest sdk, scopes are used for customer managed keys and it is not making sense in case of Customer provided keys, seems like they have clubbed it here. Let me know what u think ? 
So empty scope will always be mapped to default that is same CPK will be used for all blobs in the container.

Reply:
But customer providing scope in CPK seems out of place. As apparently you don't have access to key vault in case of CPK . Everywhere they have mentioned using scope only in Customer managed key or Microsoft managed key. 
https://learn.microsoft.com/en-us/azure/storage/blobs/encryption-scope-overview

Check this https://learn.microsoft.com/en-us/rest/api/storageservices/put-block?tabs=microsoft-entra-id
Here also the header for CPK does not have scope.